lib.js

```diff
@@ -41,36 +45,98 @@ const handleMessages = ( self, ms ) => {
 
 const reduceHtml = m => {
     const mp = m;
-    // <p>
-    //mp.message = m.message.substring( 3, m.message.length-5 );
+
+    const unescapeHtml = unsafe => {
+        return unsafe
+            .replace( /&amp;/g,  `&` )
+            .replace( /&lt;/g,   `<` )
+            .replace( /&gt;/g,   `>` )
+            .replace( /&quot;/g, `"` )
+            .replace( /&#39;/g,  `'` );
+    };
+
+    // <p></p>
+    mp.message = mp.message.replace( /<\/?p[\w =#"':\/\\.\-?]*>/gi, "" );
 
     // <a>gets left</a>
-    mp.message = mp.message.replace( /<\/?a[\w ="':\/\\.]*>/gi, "" );
+    // Custom links are text in <a>, and then the link in <kbd>
+    mp.message = mp.message.replace( /<\/?a[\w -=#"':\/\\.\-?]*>/gi, "" );
+    mp.message = mp.message.replace( "<kbd>", "" );
+    mp.message = mp.message.replace( "</kbd>", "" );
 
     // <img> to :emotes:
-    //mp.message = mp.message.replace( /<img[\w ="':\/\\.\-?]*>/gi, m => {
-    //    return m.search( /https:\/\/[\/\w\.\-?]*/ );
-    //});
+    mp.message = mp.message.replace( /<img[\w -=#"':\/\\.\-?]*>/gi, m => {
+        // (((, ))), :), :( are rendered differently
+        // (They dont even show up in the emote list)
+        //
+        // It wouldn't be so bad if the two echos were 'echol' and 'echor', but
+        //  one echo is flipped with CSS.
+        if( m.includes('alt="echo"') ) {
+            return m.includes('scaleX(-1)') ? "(((" : ")))";
+        }
+        return m.match( /alt="([\w:()]+)"/ )[1];
+    });
+
+    mp.message = turndownService.turndown( mp.message );
+
     return mp;
 };
 
 const roomCheck = ( m ) => {
     return ( m.channel !== config.room && !config.global ) ? undefined : m;
 };
 
+const commandParserSettings = {
+    prefix: '!',
+    commands: new Map(),
+};
+const isCommand = m => {
+    return m.message.startsWith( commandParserSettings.prefix ) ? m : false;
+};
+const commandParser = m => {
+    // removes prefix
+    m.message = m.message.replace( commandParserSettings.prefix, "" );
+    const args = m.message.split(' ');
+
+    const command = commandParserSettings.commands.get(args[0]);
+    if( command ) {
+        command( m, args.splice(1, 1) );
+    } else {
+        $log.info( `No command ${args[0]} found` );
+    }
+};
+
 export default {
 
     get config() { return config; },
     set config(c) { config = c; },
 
+    get commandParserSettings() { return commandParserSettings; },
+    set commandParserSettings(s) { commandParserSettings = s; },
+
     transformers: [ reduceHtml ],
-    filters: [ roomCheck ],
+    filters: [ roomCheck, isCommand ],
 
-    consumer( m ) { console.log( m ); },
+    consumer( m ) { commandParser( m ); console.log( m ); },
```

Comment:
```suggestion
    consumer( m ) { console.log( m ); commandParser( m ); },
```
The `commandParser` function takes off the prefix before returning it, so the `console.log` displays the trimmed message.